airbyte-integrations/connectors/source-shopify/unit_tests/integration/api/bulk.py

```diff
@@ -0,0 +1,177 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import json
+from datetime import datetime
+from random import randint
+
+from airbyte_cdk.test.mock_http import HttpRequest, HttpResponse
+
+
+def _create_job_url(shop_name: str) -> str:
+    return f"https://{shop_name}.myshopify.com/admin/api/2024-04/graphql.json"
+
+
+def create_job_creation_request(shop_name: str, lower_boundary: datetime, upper_boundary: datetime) -> HttpRequest:
+    outer_query = """mutation {
```

Comment:
This could be defined as follows:
```python
# add missing imports 
from source_shopify.shopify_graphql.bulk.query import ShopifyBulkTemplates

# build the query
query = """{
  orders(
    query: \"updated_at:>='%LOWER_BOUNDARY_TOKEN%' AND updated_at:<='%UPPER_BOUNDARY_TOKEN%'\"
    sortKey: UPDATED_AT
  ) {
    edges {
      node {
        __typename
        id
        metafields {
          edges {
            node {
              __typename
              id
              namespace
              value
              key
              description
              createdAt
              updatedAt
              type
            }
          }
        }
      }
    }
  }
}"""

prepared_query = ShopifyBulkTemplates.prepare(query)
```

Reply:
We can also build the actual query using the constructor, instead of the `str` value. I realize the string value is easy to go with, but it's also easy to lose when debugging staff.

Reply:
We agreed synchronously that it is a balance between test maintainability/readability and having a better safety net. For now, we will prioritize the former over the latter and document this decision. If an issue in prod allows us to challenge this balance, the dev in charge should have all the information to revert this decision if needed.